src/ngMessages/messages.js

```diff
@@ -496,31 +496,16 @@ angular.module('ngMessages', [])
     * @param {string} ngMessagesInclude|src a string value corresponding to the remote template.
     */
    .directive('ngMessagesInclude',
-     ['$templateRequest', '$document', '$compile', function($templateRequest, $document, $compile) {
+     ['$templateRequest', '$compile', function($templateRequest, $compile) {
 
      return {
        restrict: 'AE',
-       require: '^^ngMessages',
-       compile: function(element, attrs) {
-         var comment = jqLite($document[0].createComment(' ngMessagesInclude: '));
-         element.after(comment);
-
-         return function($scope, $element, attrs, ngMessagesCtrl) {
-           // we're removing this since we only need access to the newly
-           // created comment node as an anchor.
-           element.remove();
-
-           $templateRequest(attrs.ngMessagesInclude || attrs.src).then(function(html) {
-             var elements = jqLite('<div></div>').html(html).contents();
-             var cursor = comment;
-             forEach(elements, function(elm) {
-               elm = jqLite(elm);
-               cursor.after(elm);
-               cursor = elm;
-             });
-             $compile(elements)($scope);
-           });
-         };
+       require: '^^ngMessages', // we only require this for validation sake
+       link: function($scope, element, attrs) {
+         $templateRequest(attrs.ngMessagesInclude || attrs.src).then(function(html) {
+           element.html(html);
+           $compile(element.contents())($scope);
```

Comment:
What's the benefit to having this code be inserted in a different way? There's no animation stuff here since that's managed in the `<ng-message>` directive level which transcludes on its own anyway.